Assets/Scripts/HandController.cs

```diff
@@ -4,36 +4,53 @@
 
 public class HandController : MonoBehaviour
 {
+    public List<GameObject> cards;
+    public List<GameObject> selectedCards;
+    public int maxCards;
     public GameObject cardPrefab;
-    public GameObject[] cards;
-    public int numCards;
 
-    // Start is called before the first frame update
+    public CastButtonController castButton;
+
     void Start()
     {
         GenerateCards();
     }
 
+    public void SelectCard(GameObject card) {
+        if (!selectedCards.Contains(card)) {
+            selectedCards.Add(card);
+        } else {
+            selectedCards.Remove(card);
+        }
+
+        castButton.OnSelectionChanged(selectedCards.Count);
+    }
+
     void GenerateCards() {
-        for (int i = 0; i < numCards; i++) {
+        for (int i = 0; i < maxCards; i++) {
             GenerateCard(i);
         }
     }
 
     void GenerateCard(int index) {
-            string cardName = string.Format("card{0}", index);
-            Vector3 cardTransform = alignCard(index+1, (RectTransform)transform);
-            
-            GameObject cardGameObject = Instantiate(cardPrefab, cardTransform, transform.rotation);
-            cardGameObject.name = cardName;
+        string cardName = string.Format("card{0}", index);
```

Comment:
Can we do "Card-{0}"? small nit but I think it's prettier and is more consistent

Reply:
Sure. Changed.